algoliasearch/types_rule.go

```diff
@@ -1,11 +1,35 @@
 package algoliasearch
 
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
 type Rule struct {
-	ObjectID        string          `json:"objectID,omitempty"`
-	Condition       RuleCondition   `json:"condition"`
-	Consequence     RuleConsequence `json:"consequence"`
-	Description     string          `json:"description,omitempty"`
-	HighlightResult Map             `json:"_highlightResult,omitempty"`
+	Condition            RuleCondition   `json:"condition"`
+	Consequence          RuleConsequence `json:"consequence"`
+	Description          string          `json:"description,omitempty"`
+	Enabled              bool            `json:"enabled"` // Defaults to true
+	HighlightResult      Map             `json:"_highlightResult,omitempty"`
+	ObjectID             string          `json:"objectID,omitempty"`
+	Validity             []TimeRange     `json:"validity,omitempty"`
+	isExplicitlyDisabled bool
+}
+
+func (r *Rule) Enable() {
+	r.Enabled = true
+}
+
+func (r *Rule) Disable() {
```

Comment:
After discussing it, I guess having these `Enable` / `Disable` methods is the best way to handle it. We'll need to document it tho.